homeassistant/components/livisi/shc.py

```diff
@@ -111,52 +57,17 @@ async def async_setup(self) -> None:
                 livisi_connection_data=livisi_connection_data
             )
         controller_info = await self.aiolivisi.async_get_controller()
-        if bool(controller_info) is False:
-            controller_info = await self.aiolivisi.async_get_controller()
 
         controller_data = controller_info
         if controller_data.get("controllerType") == "Avatar":
-            self._port = AVATAR_PORT
-            self._is_avatar = True
+            self.port = AVATAR_PORT
+            self.is_avatar = True
         else:
-            self._port = CLASSIC_PORT
-            self._is_avatar = False
+            self.port = CLASSIC_PORT
+            self.is_avatar = False
         self.serial_number = controller_data.get("serialNumber")
```

Comment:
This looks like it can return `None`. That doesn't match the type annotation of `serial_number`.